ledger/nomos-ledger/src/mantle/sdp/mod.rs

```diff
@@ -212,14 +205,30 @@ impl SessionState {
         }
         self.clone()
     }
+
+    /// Get provider ID by index using the stable iteration order of the map.
+    ///
+    /// The `RedBlackTrieMapSync` provides a stable iteration order based on its
+    /// internal structure.
```

Comment:
Is this order insertion order? or ordered by the ord defined on `ProviderId`?

Reply:
order by `ProviderId`